qiskit_nature/drivers/second_quantization/base_driver.py

```diff
@@ -78,9 +63,8 @@ def molecule(self) -> Optional[Molecule]:
     @molecule.setter
     def molecule(self, value: Molecule) -> None:
         """set molecule"""
-        if not self.supports_molecule:
-            raise QiskitNatureError("Driver doesn't support molecule.")
         self._molecule = value
+        self._supports_molecule = self.molecule is not None
```

Comment:
I don't really understand the point of this line..
Do we even still need `supports_molecule` or can we simply rely on whether `from_molecule` is implemented?

Reply:
Fixed supports_molecule as the above was bogus. Since other methods in Nature use argument of type BaseDriver and access the molecule property, it is still needed so that they fail when try to access the property for a driver that doesn't support it.